### ISSUE TITLE
AssertionHelper requires specific NUnit version

### ISSUE THREAD

**Issue by rprouse** [nunit/NUnit.StaticExpect#5]
This project references a specific version of the NUnit framework. Because we strong sign the framework, that will mean that we will need to release a copy of this with every release of NUnit which is something I would prefer not to do. https://github.com/nunit/nunit/pull/2563/files directs users to using this project as part of the obsolete.

Users will also have to fully qualify the `AssertHelper` or it will conflict with the version still in the framework.

So, @nunit/core-team and @nunit/framework-team what do you think we should do? 

1. Is there a way to require any version of NUnit 3.x+? We could use reflection to load the asserts, but that seems to be a PITA.
2. Is anyone willing to release this project with every release of the framework? I'm not.
3. Should we just drop this and remove it from the obsolete message? Eventually we will remove `AssertionHelper` from the framework. At 4.0?

Personally, I'm in favor of option 3.

**Comment by CharliePoole** [comment]
This is a project I created and which I believe we decided not to release because we chose a different approach. IIRC, we were going to refer people to a separate third-party project.

I think the code is still here only out of an abundance of caution.

**Comment by rprouse** [comment]
@CharliePoole that may be correct, but the `Obsolete` attribute that is currently in the code points to a non-existent library, so we need to release this or remove the mention in the framework. That was one of those issues that is so long it is hard to know exactly what we decided 😄 

As I said, I just favour removing the mention in the framework.

**Comment by ChrisMaddock** [comment]
I believe the ObsoleteAttribute was intended to point to the version of this project that @fluffynuts maintains over here: https://github.com/fluffynuts/NUnit.StaticExpect

If I've remembered correctly - I think we should archive this version, to avoid confusion!

**Comment by jnm2** [comment]
IIRC strong-naming isn't a problem because the build system can auto-generate the binding redirects.
For test projects, this currently means adding two lines to the csproj. `https://github.com/dotnet/sdk/issues/1351`

**Comment by CharliePoole** [comment]
@rprouse Not non-existent, as @ChrisMaddock points out, just not ours. We didn't have a repository of that name until just now.

IMO, you could delete this repository rather than archive it. It was a proposal that the team didn't accept and it was never released anywhere. Although I created it as a proposal, I didn't feel I should delete it without some final team decision.

**Comment by fluffynuts** [comment]
It seems as if the general sentiment has always been to eventually remove `AssertionHelper` from `NUnit`. The current userbase of `NUnit.StaticExpect` is not overwhelming -- I see 149 downloads for the current release. However, I'm happy to maintain it for as long as people find it even remotely useful.
Some suggestions come to mind:
1. `AssertionHelper` isn't changing, so for the moment, assembly rebinding takes care of the fact that `NUnit` versions supersede the version against which `NUnit.StaticExpect` is built. So this could be left alone until the time when `AssertionHelper` is to be officially removed, at which point, I could fork `NUnit`, reference via git submodule, and build with `AssertionHelper` internal to `NUnit.StaticExpect`. Assembly rebinds are still going to happen in the future though, since the point would be to use the version of `NUnit` the user chooses. It should be noted that the current package dependency set up in `NUnit.StaticExpect`'s nuspec causes `NUnit` 3.8.1 to be installed if `NUnit.StaticExpect` is installed first. The user can upgrade, but ideally, the user should get the latest `NUnit` version, up to, but not including, 4. I'm making a change to the package.nuspec to try to affect this, according to the documentation here: https://docs.microsoft.com/en-us/nuget/reference/nuspec
2. I could release `NUnit.StaticExpect` in lock-step with `NUnit`. I don't mind -- it's not a lot of effort, really -- it would just have to lag behind `NUnit`, out of necessity, by up to 24 hours. This really only affects assembly rebinding, since `AssertionHelper` remains unchanged through `NUnit` releases. 
3. I could absorb `AssertionHelper` into `NUnit.StaticExpect` now, as per (1) above. I'm not sure of the amount of work involved, but I assume it to be possible -- I use a similar trick with `NExpect`, to use utilities out of my `PeanutButter` project where I don't want `NExpect` users to have to reference a bunch of other assemblies they potentially have no interest in. This would free the `NUnit` team to remove AssertionHelper whenever it's convenient, without any concern for `NUnit.StaticExpect` users. This only makes sense if `AssertionHelper` is properly frozen at this point, which I assume it is.

Thoughts? As always, I'm interested in whatever serves the users and the NUnit team's purposes best. I don't have much of a dog in this fight as my personal usage has shifted completely to `NExpect`, but I'm always willing to maintain whilst there are people who want the functionality.

**Comment by rprouse** [comment]
Thanks for the updates. Clearly I was confused 😕 

Based on the fact that it does exist, shall we delete this repo to prevent confusion (like mine) and leave the framework as-is?

**Comment by rprouse** [comment]
Also, should @fluffynuts absorb `AssertionHelper` now? If so, he could just grab the one file from here along with the tests. @fluffynuts you may want to do that on a branch in preparation. We won't be changing `AssertionHelper` in the framework.

**Comment by ChrisMaddock** [comment]
I agree we should delete this repo down, and I think @fluffynuts releasing NUnit.StaticExpect in lock-step is sufficient for the time being.

**Comment by fluffynuts** [comment]
@rprouse, btw, the issue this is linked to (https://github.com/nunit/nunit/issues/2646) shouldn't have any impact -- I don't rely on the zip package, rather I set up my nuspec to depend on `NUnit`. Anyways, I'll go build a version now which explicitly targets `NUnit` latest as the minimum version for dependency.
